app/_components/MapContainer.tsx

```diff
@@ -91,6 +98,16 @@ function MapInner() {
     handleDistanceToClosestPin(userCoordinates, closestNotCompletedPin);
   }, [closestNotCompletedPin, userCoordinates]);
 
+  //after each guess
+  useEffect(() => {
+    void handleLevelAndXp();
+  }, [checkLevel]);
+
+  //NOTE: upon first load, we should refacor BE so that the Data gets send when loggin in.
+  useEffect(() => {
+    void handleLevelAndXp();
+  }, []);
+
```

Comment:
With how useEffect work, it will do every useEffect by the initial fetch. so no need for duplicate useEffect without dependencies. it will slow down the initial fetch and can cause some bug since the handler function will trigger twice when first load. also can we move this into the `LevelProgressbar.tsx` since it has no relation to another logic and function in MapContainer.tsx

```suggestion
  //after each guess
  useEffect(() => {
    void handleLevelAndXp();
  }, [checkLevel]);
```